R/run.R

```diff
@@ -0,0 +1,106 @@
+#' the output metadata.
+#' @param returnRaw Boolean to control the return type. By default (returnRaw = FALSE), run() returns a named list
+#' result, where result$df is a data frame which contains the result in the same format as the input scenarios, and
+#' result$metadata is a named list of metadata generated during the run. If returnRaw is TRUE, run() returns a python
+#' scmdata.ScmRun object.
+#' @author Mika Pflüger
+#' @importFrom reticulate import py_to_r
+#' @examples
+#' \dontrun{
+#' # create very minimal emissions scenario.
+#' df <- data.frame(
+#'  model = c("rand", "rand"),
+#'  scenario = c("weirdEMI", "weirdEMI"),
+#'  region = c("World", "World"),
+#'  variable = c("Emissions|CO2", "Emissions|CH4"),
+#'  unit = c("Mt CO2 / yr", "Mt CH4 / yr"),
+#'  "2015" = c(9., 12.),
+#'  "2020" = c(10., 11.),
+#'  check.names = FALSE)
+#'
+#' # simulate the scenario using MAGICC7 with default settings.
+#' run(climateModelsConfigs = list("MAGICC7" = NULL), scenarios = df)
+#'
+#' # simulate the scenario using MAGICC7 with default settings but "somesetting" set to "12"
+#' run(climateModelsConfigs = list("MAGICC7" = list("somesetting" = "12")), scenarios = df)
+#'
+#' # simulate the scenario with MAGICC7 and FaIR, each with two different sets of configurations
+#' # where in MAGICC7 we change "somesetting" and in FaIR we change "fairsetting".
+#' # Also include the changed configuration settings in the output.
+#' run(climateModelsConfigs = list("MAGICC7" = list(list("somesetting" = "12"),
+#'                                                  list("somesetting" = "13")),
+#'                                 "FaIR" = list(list("fairsetting" = "slr"),
+#'                                               list("fairsetting" = "noslr"))),
+#'     scenarios = df,
+#'     outConfig = list("MAGICC7" = list("somesetting"), "FaIR" = list("fairsetting")))
+#' }
+#' @export
+run <- function(climateModelsConfigs, scenarios, outputVariables = list("Surface Temperature"), outConfig = NULL,
+                returnRaw = FALSE) {
+  openscmRunner <- import("openscm_runner", convert = FALSE)
+  scmdata <- import("scmdata")
+  sr <- scmdata$ScmRun(scenarios)
+  climateModelsConfigs <- expandConfigs(climateModelsConfigs)
+
+  if (!is.null(outConfig)) {
+    for (model in names(outConfig)) {
+      if (!is.null(outConfig[[model]])) {
+        outConfig[[model]] <- reticulate::tuple(outConfig[[model]])
+      }
+    }
+  }
+
+  rawResult <- openscmRunner$run(
+    climate_models_cfgs = climateModelsConfigs,
+    scenarios = sr,
+    output_variables = outputVariables,
+    out_config = outConfig
+  )
+  if (returnRaw) {
+    return(rawResult)
+  } else {
+    return(list(df = py_to_r(rawResult$timeseries()$reset_index()),
+                metadata = py_to_r(rawResult$metadata)))
+  }
+}
+
+expandConfigs <- function(climateModelsConfigs) {
+  emptyNamedList <- list()
+  names(emptyNamedList) <- list()
```

Comment:
I don't think this is wrong or breaks, but this still looks strange to me. Usually `names` are a character vector, not a list. So maybe this is better?
```suggestion
  names(emptyNamedList) <- character(0)
```